src/base/QXmppFutureUtils_p.h

```diff
@@ -215,6 +216,26 @@ auto mapSuccess(std::variant<T, Err> var, Function lambda)
                       std::move(var));
 }
 
+template<typename T, typename Err>
+auto mapToSuccess(std::variant<T, Err> var)
+{
+    return mapSuccess(std::move(var), [](T) {
+        return Success();
+    });
+}
+
+template<typename Input>
+auto chainSuccess(QXmppTask<Input> &&source, QObject *context) -> QXmppTask<QXmppClient::EmptyResult>
+{
+    QXmppPromise<QXmppClient::EmptyResult> promise;
```

Comment:
better use `std::variant<QXmpp::Success, QXmppError>` directly and avoid the additional include of `QXmppClient.h`